File: Data Structures & Algorithms/top-k-elements-in-list/submission-9.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:


        hashMap = {}

        count = [[] for i in range(len(nums) + 1)]


        for num in nums:
           hashMap[num] = 1 + hashMap.get(num,0) 


        for key,val in hashMap.items():

            count[val].append(key)

    
        res = []


        for i in range(len(count) - 1, -1 , -1):

            for num in count[i]:
                if k == 0:
                    return res
                res.append(num)
                k-=1
        
        return res
            
        






        

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-9.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {

        vector<int> res(nums.size());

        for ( int i = 0 ; i < nums.size() ; ++i){
            if(i == 0 ){
                res[i] = 1;
                continue;
            }

            res[i] = nums[i - 1 ] * res[i -1];
        }

        int suffix = 1;
        for( int i = nums.size() -1    ; i >= 0 ; --i){
            res[i] *=suffix;
            suffix*=nums[i]; 
        }

        return res;
    }
};
